### ISSUE TITLE
OCIError: ORA-00932: inconsistent datatypes: expected NUMBER got DATE in Rails 4.2.0

### ISSUE THREAD

**Issue by rootatdarkstar** [rsim/oracle-enhanced#537]
Stuck with this problem while i try to migrate on rails 4.2.0 and oracle_enhanced from branch rails42

Oracle enhanced quotes NUMBER value as Date type
OCIError: ORA-00932: inconsistent datatypes: expected NUMBER got DATE: 
```sql
SELECT "S_USER_REPORTS".* FROM "S_USER_REPORTS"
 WHERE ("S_USER_REPORTS"."ID_A_USERS" = TO_DATE('24','YYYY-MM-DD HH24:MI:SS') OR 
"S_USER_REPORTS"."SORACLELOGIN" = 'somelogin' AND "S_USER_REPORTS"."SISVISIBLE" = 'Y') 
ORDER BY "S_USER_REPORTS"."DREQUESTDT" DESC
```

My query:
```ruby
table = arel_table
where( table[:id_a_users].eq(User.current.id)
      .or(table[:soraclelogin].eq(User.current.db_schema).and( table[:sisvisible].eq('Y')) )
    )
```
My model:
```ruby
# Table name: S_USER_REPORTS
#
#  id            :integer          not null, primary key
#  soraclelogin  :string(255)
#  id_a_users    :integer          not null
#  drequestdt    :datetime         not null
#  dcreateddt    :datetime
#  sisvisible    :boolean          not null
#  sreportname   :string(255)      not null
#  sreportfile   :string(255)
#  sreporturl    :string(1000)
#  sreportstatus :string(255)
#  sreporttype   :string(20)
#
class UserReport < ActiveRecord::Base
  self.table_name = 'S_USER_REPORTS'
  self.primary_key = :id
  self.sequence_name = :autogenerated
  set_integer_columns :id_a_users
end
```
Db column is a NUMBER:
```sql
"ID_A_USERS" NUMBER NOT NULL ENABLE
```

Inspecting columns_hash - got wrong @cast_type:
```ruby
"id_a_users"=>
  #<ActiveRecord::ConnectionAdapters::OracleEnhancedColumn:0x00000006dc6e68
   @cast_type=
    #<ActiveRecord::Type::Date:0x00000006dd42e8
     @limit=nil,
     @precision=nil,
     @scale=nil>,
   @default=nil,
   @default_function=nil,
   @forced_column_type=:integer,
   @name="id_a_users",
   @null=false,
   @object_type=false,
   @returning_id=false,
   @sql_type="NUMBER",
   @table_name="S_USER_REPORTS",
   @virtual=false>,
```
Problem occures on:

ruby 2.1.5 MRI
oracle 11g
rails 4.2.0 and all betas and rc's
oci8 - 2.1.0 and master branch
oracle_enhanced - rails42 branch

I have no problem with rails 4.1.x and oracle_enhanced 1.5.5 from rubygems

**Comment by yahonda** [comment]
Thanks for testing rails42 branch and reporting.
I think it is caused by (kind of known) regressions introduced by #498 6ea13970

Will have a look at these later, of course pull requests are welcomed.

**Comment by pandar1** [comment]
Same problem here, was ok with 4.1.6 and broken with 4.2.0, solved using branch rails42

**Comment by yahonda** [comment]
@rootatdarkstar Would you give it a try if https://github.com/yahonda/oracle-enhanced/commit/1d505748c3f993c19d7952faa8963d103130b046 works for you by changing your `Gemfile` as follows.

```ruby
gem 'activerecord-oracle_enhanced-adapter', github: 'yahonda/oracle-enhanced', branch: 'rails42_537'
```

**Comment by rootatdarkstar** [comment]
@yahonda  Sorry for long waiting - i'm wasmissed some notifications from github.
Yes, problem was fixed. Very thank you.

**Comment by yahonda** [comment]
@rootatdarkstar Thanks for the verification.

**Comment by yahonda** [comment]
Closed since #538 has been merged to rails42
